packages/@aws-cdk/aws-cognito/lib/user-pool-client.ts

```diff
@@ -407,6 +422,37 @@ export class UserPoolClient extends Resource implements IUserPoolClient {
     return this._userPoolClientName;
   }
 
+  public get userPoolClientSecret(): string | undefined {
+    if (!this._generateSecret) {
+      throw new Error(
+        'userPoolClientSecret is available only if the "generateSecret" prop is set to true',
+      );
+    }
+    const existingCustomResource = this.node.tryFindChild('DescribeCognitoUserPoolClient') as AwsCustomResource | undefined;
+    const describeCognitoUserPoolClient = existingCustomResource ?? new AwsCustomResource(
```

Comment:
Nicely done. It should be possible to use a typed private member variable to hold this reference without the type laundering, but I don't have strong opinions here.

Reply:
Cool suggestion. Refactored to implement it...